src/Dev/DevelopmentAdmin.php

```diff
@@ -190,6 +176,33 @@ protected static function get_links()
         return $links;
     }
 
+    protected function getLinks(): array
+    {
+        $canViewAll = $this->canViewAll();
+        $links = [];
+        $reg = Config::inst()->get(static::class, 'registered_controllers');
+        foreach ($reg as $registeredController) {
+            if (isset($registeredController['links'])) {
+                if (!ClassInfo::exists($registeredController['controller'])) {
+                    continue;
+                }
+
+                if (!$canViewAll) {
+                    // Check access to controller
+                    $controllerSingleton = singleton($registeredController['controller']);
+                    if (!$controllerSingleton->hasMethod('canInit') || !$controllerSingleton->canInit()) {
+                        continue;
+                    }
```

Comment:
```suggestion
                    if ($controllerSingleton->hasMethod('canInit') && !$controllerSingleton->canInit()) {
                        continue;
                    }
```

I think if there's no `canInit()` method we should assume that access is allowed, rather than the opposite. If someone has implemented their own custom dev controller, they may have permission checks in `init()` which we're currently ignoring with this implementation.
I think.
Feel free to let me know if I've missed something there.

Reply:
The problem with this interpretation is that now anyone can see those listed at `/dev` where before it was limited to `ADMIN` - I'm not super into either interpretation, to be honest, but I erred towards secure by default

Reply:
Fair point. Admin will still be able to see them so there's no scenario where a link _was_ visible before and _won't_ be anymore. I think you're right, this is the better outcome.